packages/pg-pool/index.js

```diff
@@ -244,7 +244,7 @@ class Pool extends EventEmitter {
         // remove the dead client from our list of clients
         this._clients = this._clients.filter((c) => c !== client)
         if (timeoutHit) {
-          err.message = 'Connection terminated due to connection timeout'
+          err = new Error('Connection terminated due to connection timeout', { cause: err } )
```

Comment:
Node.js support for `Error.cause` added in 16.9.0, prior versions will silently ignore - has the effect of losing the stack trace of the original exception under these older Node.js versions